src/cmd-build

```diff
@@ -79,10 +92,36 @@ if [ -f "${changed_stamp}" ]; then
     cp -a --reflink=auto ${composejson} ${workdir}/tmp/compose-${commit}.json
 else
     commit=${previous_commit}
-    # Grab the previous JSON
-    cp -a --reflink=auto ${workdir}/tmp/compose-${previous_commit}.json ${composejson}
+    image_input_checksum=$((echo ${commit} && echo ${kickstart_checksum}) | sha256sum_str)
+    if [ "${image_input_checksum}" = "${previous_image_input_checksum}" ]; then
+        echo "No changes in image inputs."
+        exit 0
+    fi
+
+    # Grab the previous treecompose JSON (local developer case: treecompose succeeded but
+    # image build failed) if possible, otherwise grab the previous build
+    cached_previous_composejson=${workdir}/tmp/compose-${previous_commit}.json
```

Comment:
seems like maybe we should print a log statement here saying we are trying to "resume the last build" or something